### PR TITLE
mgr/dashboard: Downstream branding

### PR DESCRIPTION
**Pull request by votdev**
Adds the downstream branding to the ceph dashboard.

- New sign in page
- New about dialog
- Changed color scheme
- Replaced logos

Update on 04.09.2020 - Fixed bz:1175169, bz:1175161
Update on 25.09.2020 - Fixed bz:1175120

Signed-off-by: Sebastian Krah <skrah@suse.com>
(cherry picked from commit 7563f3305a12f155c5d171e93434f1372b977ae9)

Conflicts:
  src/pybind/mgr/dashboard/frontend/src/app/shared/services/doc.service.ts
  src/pybind/mgr/dashboard/frontend/src/app/shared/services/doc.service.spec.ts
  src/pybind/mgr/dashboard/frontend/src/styles/vendor.overrides.scss
  src/pybind/mgr/dashboard/frontend/src/styles/vendor.variables.scss

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by votdev** [comment]
Questions:
Where to redirect the following URLs:

https://github.com/SUSE/ceph/pull/436/files#diff-98496ff59419b0ec12993711a530c660967874a56196916b46ae78c4d4f21436R39-L43

**Comment by votdev** [comment]
> Questions:
> 
> How shall we map this links? Do we have any equivalent in our documentation?
> 
>     * [ceph.io/pgcalc](https://ceph.io/pgcalc)
> 
>     * [ceph.io/help](https://ceph.io/help/)
> 
>     * [ceph.io/security](https://ceph.io/security/)
> 
>     * [ceph.io/legal-page/trademarks](https://ceph.io/legal-page/trademarks/)
> 
> 
> [https://github.com/votdev/ceph/blob/ses7p_branding/src/pybind/mgr/dashboard/frontend/src/app/shared/services/doc.service.ts#L39-L42](https://github.com/votdev/ceph/blob/ses7p_branding/src/pybind/mgr/dashboard/frontend/src/app/shared/services/doc.service.ts?rgh-link-date=2021-07-21T14%3A57%3A34Z#L39-L42)

After a discussion with Tomas we will drop the `pgcalc`, `security` and `trademark` links because we do not have equivalents.

**Comment by smithfarm** [comment]
> How shall we map this links? Do we have any equivalent in our documentation?
> 
>     https://ceph.io/pgcalc
>     https://ceph.io/help/
>     https://ceph.io/security/
>     https://ceph.io/legal-page/trademarks/

Three of these four URLs are broken (404)...

**Comment by votdev** [comment]
> > How shall we map this links? Do we have any equivalent in our documentation?
> > ```
> > https://ceph.io/pgcalc
> > https://ceph.io/help/
> > https://ceph.io/security/
> > https://ceph.io/legal-page/trademarks/
> > ```
> 
> Three of these four URLs are broken (404)...

Yes, i know. I've mapped them like this:

https://ceph.io/pgcalc -> https://documentation.suse.com/ses/7p/single-html/ses-admin/#ceph-pools
https://ceph.io/help/ -> https://documentation.suse.com/ses/7p/single-html
https://ceph.io/security/ -> https://documentation.suse.com/ses/7p/single-html/ses-security
https://ceph.io/legal-page/trademarks/ -> https://documentation.suse.com/ses/7p/single-html

**Review by s0nea — COMMENTED** [review]
I tested the PR with **Firefox 78.12.0esr** and **Chrome 91.0.4472.114**.

A few things I noticed until now:

The closing 'x' in the top right corner of the about modal isn't visible:

Screenshot_2021-07-30_16-27-30

I'm not sure if I remember it correctly but did we replace the Ceph icon by Geeko in the past?

Screenshot_2021-07-30_16-19-53

Screenshot_2021-07-30_16-49-16

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.brand.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<div *ngIf="isLoginActive">
+  <h2 i18n>Login</h2>
+  <form name="loginForm"
+        (ngSubmit)="login()"
+        #loginForm="ngForm"
+        novalidate>
+
+    <!-- Username -->
```
[the remark this answers is not in the document]
Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.brand.html:8]: This also happens in the SES7 version. Need to fix that for Chrome based browsers.

[the remark this answers is not in the document]
Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.brand.html:8]: Fixed

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/doc.service.ts [hunk 2]
```diff
@@ -23,27 +23,24 @@ export class DocService {
...
-      trademarks: `${domainCeph}legal-page/trademarks/`,
-      'dashboard-landing-page-status': `${domain}mgr/dashboard/#dashboard-landing-page-status`,
-      'dashboard-landing-page-performance': `${domain}mgr/dashboard/#dashboard-landing-page-performance`,
-      'dashboard-landing-page-capacity': `${domain}mgr/dashboard/#dashboard-landing-page-capacity`
+      iscsi: `${domain}ses-admin/#dashboard-iscsi-management`,
+      prometheus: `${domain}ses-deployment/#deploy-cephadm-day2-service-monitoring`,
+      'nfs-ganesha': `${domain}ses-admin/#ceph-nfsganesha-config`,
+      'rgw-nfs': `${domain}ses-admin/#ceph-nfsganesha-config-service-rgw`,
+      rgw: `${domain}ses-admin/#dashboard-ogw-enabling`,
+      dashboard: `${domain}ses-admin/#dashboard-initial-configuration`,
```
[the remark this answers is not in the document]
Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/services/doc.service.ts:34]: Fixed

**Comment by votdev** [comment]
> I tested the PR with **Firefox 78.12.0esr** and **Chrome 91.0.4472.114**.
> 
> A few things I noticed until now:
> 
> The closing 'x' in the top right corner of the about modal isn't visible:
> 
> Screenshot_2021-07-30_16-27-30
> 
> I'm not sure if I remember it correctly but did we replace the Ceph icon by Geeko in the past?
> 
> Screenshot_2021-07-30_16-19-53
> 
> Screenshot_2021-07-30_16-49-16

Fixed

**Review by s0nea — APPROVED** [review]
Thanks for addressing my comments. LGTM :+1:

**Comment by smithfarm** [comment]
@votdev Please merge when you think this is ready.

**Pull request merged**